src/Application/Group/Command/RenameGroupHandler.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the `gata` project.
+ *
+ * (c) Aula de Software Libre de la UCO <aulasoftwarelibre@uco.es>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Application\Group\Command;
+
+use App\Application\Group\Exception\GroupNotFoundException;
+use App\Application\Group\Repository\Groups;
+use App\Domain\Group\Model\Group;
+
+final class RenameGroupHandler
+{
+    /**
+     * @var Groups
+     */
+    private $groups;
+
+    public function __construct(Groups $groups)
+    {
+        $this->groups = $groups;
+    }
+
+    public function __invoke(RenameGroup $renameGroup): void
+    {
+        $group = $this->groups->get($renameGroup->groupId());
+
+        if (!$group instanceof Group) {
```

Comment:
I guess this check should be in `\App\Infrastructure\Repository\EventStoreGroups::get()`

Reply:
La responsabilidad de comprobar si existe (guardemos de la forma que sea los grupos) o no un grupo es de la capa de aplicación o infraestructura. Igualmente, junto a esta comprobación deberíamos de movel la excepción Application/Group/Exception/GroupNotFoundException.php

Reply:
You are getting for something that does not exist. So, the exception should be in the infrastructure layer. But the exception remains in the application layer because the application layer shouldn't depends from anything from the infrastructure layer.